### PR TITLE
Update latest code generation segments

### PR DESCRIPTION
**Pull request by tuommaki**
Remove deprecated functions, bring few field copies up-to-date.

**Edit:** I was about to add a new field to `AzureConfig.Status` but noticed that code generator refreshed parts of existing code so those changes are separately here. They mostly seem fine, but please double check if those ring any bells.

### CODE REVIEW

File: pkg/apis/application/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func (in *AppCatalog) DeepCopyObject() runtime.Object {
 func (in *AppCatalogList) DeepCopyInto(out *AppCatalogList) {
 	*out = *in
 	out.TypeMeta = in.TypeMeta
-	out.ListMeta = in.ListMeta
+	in.ListMeta.DeepCopyInto(&out.ListMeta)
```
Comment (tuommaki) [inline pkg/apis/application/v1alpha1/zz_generated.deepcopy.go]: This makes sense.

File: pkg/clientset/versioned/clientset.go [hunk 2]
```diff
@@ -32,20 +32,10 @@ import (
 type Interface interface {
 	Discovery() discovery.DiscoveryInterface
 	ApplicationV1alpha1() applicationv1alpha1.ApplicationV1alpha1Interface
-	// Deprecated: please explicitly pick a version if possible.
-	Application() applicationv1alpha1.ApplicationV1alpha1Interface
```
Comment (tuommaki) [inline pkg/clientset/versioned/clientset.go]: Dropping deprecated function makes also sense.

File: pkg/clientset/versioned/fake/clientset_generated.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ func NewSimpleClientset(objects ...runtime.Object) *Clientset {
 		}
 	}
 
-	cs := &Clientset{}
+	cs := &Clientset{tracker: o}
```
Comment (tuommaki) [inline pkg/clientset/versioned/fake/clientset_generated.go]: This is something I'm not fully grasping, but it's fake client that's used for tests, so I figure it's there to improve tracking of [fake] API object mutations.

File: pkg/clientset/versioned/typed/application/v1alpha1/application_client.go [hunk 4]
```diff
@@ -81,7 +80,7 @@ func setConfigDefaults(config *rest.Config) error {
 	gv := v1alpha1.SchemeGroupVersion
 	config.GroupVersion = &gv
 	config.APIPath = "/apis"
-	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+	config.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
```
Comment (tuommaki) [inline pkg/clientset/versioned/typed/application/v1alpha1/application_client.go]: Seems legit I figure:
https://godoc.org/k8s.io/apimachinery/pkg/runtime/serializer#CodecFactory.WithoutConversion

**Review by kopiczko — APPROVED** [review]

**Comment by tuommaki** [comment]
I did quick smoke test by cloning into `aws-operator@master` vendor and compilation + unit tests ran fine.

**Pull request merged**